fx_base.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#ifdef FWK_TESTING_MODE
+
+#include <fwk/format.h>
+#include <fwk_vector.h>
+
+#undef CHECK
+
+namespace fx {
+template <class T> using vector = fwk::vector<T>; // for additional checks
+using fwk::print;
+}
+
+#else
+
+#include "debug.h"
+#include <vector>
+
+namespace fx {
+template <class T> using vector = std::vector<T>;
+}
+
+#ifndef DASSERT // It can only be defined in fx_tester
+#define DASSERT CHECK
+#define ASSERT CHECK
+#define PASSERT CHECK
+#endif
+
+#endif
+
+namespace cereal {
+template <class T> class NameValuePair;
+class JSONOutputArchive;
+class JSONInputArchive;
+}
+
+namespace fx {
+
+template <class T> using NVP = cereal::NameValuePair<T>;
+using IArchive = cereal::JSONInputArchive;
+using OArchive = cereal::JSONOutputArchive;
+
+static constexpr int default_tile_size = 24;
```

Comment:
You can get this value from Renderer::getNominalSize(). (feel free to make that method static).